### PR TITLE
feat(LayoutGrid): Convert LayoutGrid to CSS in JS WEB-968

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview

NOTE: Not considered feature complete, just exploring some options, but tagging everyone incase they wanted to see a possible approach.

Learnings:
- Definitely need to recatalog and structure gamut-styles and types for this to be scalable.
- Scalable utilities with looser types are going to be a pain, at least in the context of single context responsive properties.   I tried making something work with `responsiveProperty` but it has its own failings (which might be due to my typescript ineptitude, open to ideas if people like this pattern).
- Overall the type checking works well if of course you don't try to do fancy things.

### PR Checklist

- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: WEB-968
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Layout/utilities.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { map, entries } from 'lodash';
+import { ResponsiveProperty } from '../typings/responsive-properties';
+import { breakpoints } from '@codecademy/gamut-styles/utils/variables';
+import { MediaSize, ValidValues } from './types';
+
+export function getViewport(
+  mediaSize: MediaSize,
+  template: string | SerializedStyles
+) {
+  if (mediaSize !== 'xs') {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Layout/utilities.tsx:11]: This part confuses me: if `mediaSize` is `xs`, won't we want to just print the `template` directly?

Alternately, should `breakpoins.xs` be `0`? 😄

Comment (codecaaron) [reply packages/gamut/src/Layout/utilities.tsx:11]: Oh shoot yeah I forgot to re-add that >.> Nice catch!

Yeah I'm not sure how to solve the problem of GTE vs LTE media queries in this context.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.0-alpha.2acbc2.0<br/>@codecademy/gamut-styles@4.2.0-alpha.2acbc2.0<br/>@codecademy/gamut@12.1.0-alpha.2acbc2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1f3678074cbc183747f4d3--gamut-preview.netlify.app>https://5f1f3678074cbc183747f4d3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f1f3678074cbc183747f4d3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.2-alpha.23d922.0<br/>@codecademy/gamut@12.3.1-alpha.23d922.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1f51cc6f557912d64f9b39--gamut-preview.netlify.app>https://5f1f51cc6f557912d64f9b39--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f1f51cc6f557912d64f9b39>Deploy Logs</a></p>

**Pull request closed without merging**